utam-compiler/src/main/java/utam/compiler/grammar/ErrorProducer.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root
+ * or https://opensource.org/licenses/MIT
+ */
+package utam.compiler.grammar;
+
+import com.fasterxml.jackson.core.JsonLocation;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import utam.compiler.helpers.TranslationContext;
+import utam.core.framework.consumer.UtamError;
+
+/**
+ * This exception is thrown from deserializer when we do not know the context. Then we catch it and
+ * process
+ *
+ * @author elizaveta.ivanova
+ * @since 238
+ */
+class ErrorProducer extends RuntimeException {
```

Comment:
If we're going to be `throw`ing these objects, wouldn't it make sense for the class to be named something ending in `Exception` or `Error`?

Reply:
renamed to UtamCompilerErrorWithCode